File: src/0121-买卖股票的最佳时机/0121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
      int sell = 0;
      int buy = INT_MAX;
      for (int i = 0; i< prices.size(); i++){
        buy = min(buy, prices[i]);
        sell = max(sell, prices[i] - buy);
      }
      return sell;
    }
};

File: src/0121-买卖股票的最佳时机/0121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        min_price = prices[0]
        max_profit = 0
        for price in prices[1:]:
            # 记录最小价格
            if price < min_price:
                min_price = price
            # 记录最大利润
            if price - min_price > max_profit:
                max_profit = price - min_price
        return max_profit